doc/source/how-to/releasing.rst

```diff
@@ -0,0 +1,280 @@
+
+.. admonition:: Do not distribute artifacts without approval
+
+   Refer to :ref:`Project Approval` for more information about requesting
+   approval for publishing your project.
+
+
+Publishing to Private PyPI
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+There are cases in which it is necessary to host and pull packages that are
+not ready to be hosted to the public `PyPI`_. For example, if a PyAnsys library
+requires auto-generated gRPC interface files from an as-of-yet private feature
+or service, this package should be hosted on a private PyPI repository.
+
+ANSYS has a private repository at `PyAnsys PyPI`_, and access is controlled via
+a secret ``Personal Access Token (PAT)``, specified in the GitHub secret
+``PYANSYS_PYPI_PRIVATE_PAT`` which is only available to repositories within the
+`PyAnsys`_.
+
+.. admonition:: A note on secrets access in forked repositories
+   Forked GitHub repositories do not have access to GitHub secrets. This is
+   designed to protect against PRs that could potentially scrape tokens from
+   our CI/CD.
+
+Upload
+++++++
+Packages can be uploaded to the private repository with the following short
+bash script. If you are operating out of a GitHub CI pipeline, email the
+PyAnsys Core team at pyansys.core@ansys.com for the ``PAT``,
+``PYANSYS_PYPI_PRIVATE_PAT``.
+
+Assuming you are already in a Python repository containing your wheels and/or
+source distribution within the ``dist`` directory:
+
+.. code::
+
+   pip install build twine pip -U
+
+   REPOSITORY_URL="https://pkgs.dev.azure.com/pyansys/_packaging/pyansys/pypi/upload"
+   python -m twine upload dist/* \
+     -p ${{ secrets.PYANSYS_PYPI_PRIVATE_PAT }} \
+     -u PAT \
+     --repository-url $REPOSITORY_URL
+
+Alternatively, you can use environment variables instead of CLI arguments for twine.
+
+.. code::
+
+   export TWINE_USERNAME=PAT
+   export TWINE_PASSWORD=$PYANSYS_PYPI_PRIVATE_PAT
+   export TWINE_REPOSITORY_URL="https://pkgs.dev.azure.com/pyansys/_packaging/pyansys/pypi/upload"
+
+   python -m twine upload dist/*
+
+
+Download
+++++++++
+To download the Python package from the `PyAnsys PyPI`_, use the following:
+
+.. code::
+
+   INDEX_URL=https://$PYANSYS_PYPI_PRIVATE_PAT@pkgs.dev.azure.com/pyansys/_packaging/pyansys/pypi/simple/
+   pip install ansys-<product/tool>-<library> --index-url $INDEX_URL --no-dependencies
+
+.. note::
+   An additional read-only PAT is available for individual users who only need to download the package.
+   It can be stored as an environment variable and be used just as described in the code sample above.
+   Please email the PyAnsys Core team at pyansys.core@ansys.com to get access to this PAT.
+
+.. warning::
+   Take care to always use the ``--index-url`` switch rather than the
+   ``--extra-index-url`` switch. As noted in the `pip Documentation`_, the
+   ``--index-url`` switch changes the Python Package Index, and forces ``pip``
+   to only use packages from that package index.
```

Comment:
```suggestion
   Take care to always use the ``--index-url`` switch rather than the
   ``--extra-index-url`` switch. As noted in `pip Documentation`_, the
   ``--index-url`` switch changes the Python Package Index, which forces ``pip``
   to use only packages from that package index.
```